packages/java/src/browser/java-client-contribution.ts

```diff
@@ -44,9 +48,25 @@ export class JavaClientContribution extends BaseLanguageClientContribution {
 
     protected onReady(languageClient: ILanguageClient): void {
         languageClient.onNotification(ActionableNotification.type, this.showActionableMessage.bind(this));
+        languageClient.onNotification(StatusNotification.type, this.showStatusMessage.bind(this));
         super.onReady(languageClient);
     }
 
+    protected showStatusMessage(message: StatusReport) {
+        const statusEntry = {
+            alignment: StatusBarAlignment.LEFT,
+            priority: 1,
+            text: "$(refresh~spin) " + message.message
+        } as StatusBarEntry;
+        this.statusBar.setElement(this.statusNotificationName, statusEntry);
+        if (!this.statusBarTimeout) {
+            this.statusBarTimeout = window.setTimeout(() => {
+                this.statusBar.removeElement(this.statusNotificationName);
+                this.statusBarTimeout = undefined;
+            }, 3000);
```

Comment:
while testing, I've noticed, that this might be not enough to wait for a possible continuation. I've tried with 5sec, and haven't seen it flicker.